client/src/views/workspaces/WorkspaceSharingModal.vue

```diff
@@ -310,6 +461,72 @@ async function updateUserRole(
   }
   await refreshSharingInfo();
 }
+
+async function onBatchRoleChange(newRoleOption: MsOption): Promise<void> {
+  let changesMade: boolean = false;
+  let errorOccured: boolean = false;
+
+  for (const userRole of selectedUsers.value) {
+    if (!userRole.role || newRoleOption.key !== userRole.role) {
+      const result = await shareWorkspace(props.workspaceId, userRole.user.id, newRoleOption.key);
+      if (result.ok) {
+        changesMade = true;
+      } else {
+        errorOccured = true;
+      }
+    }
+  }
+
+  if (!changesMade) {
```

Comment:
Error handling is insufficient here. 
You can test it in web mode by updating the mocks for `shareWorkspace` and add (as an example)
`return { ok: false, error: { tag: ClientShareWorkspaceErrorTag.Offline, error: 'Offline' } };`

The error I get is `Selected members were already Contributor.`, which is false. I'm not even sure that this error is possible, if you try to assign the same role to a user, I think it'll just work.

Handling multiple errors is always a pain. The best thing to do imo is to count the number of successes, the number of errors, and store the first error (if any). You can then compare it to the number of operations: 

- `successesCount === selectedUsers.value.length` means all good
- `errorsCount === selectedUsers.value.length` means all failed, probably a very general case like being offline, being revoked, having insufficient permission (you can check using the error you stored and assume it's the same for all)
- some errors is a complicated case, but a generic error like "some failed" is probably enough